docs/source/advancedcommands.rst

```diff
@@ -1,6 +1,117 @@
+
+
+Sweep Substitutions
+-------------------
+Alternatively, or to sweep in multiple dimensions simultaneously, Variables can swept with a substitution value takes the form ``('sweep', Iterable), (e.g. 'sweep', np.linspace(1e6, 1e7, 100))``. During variable declaration, giving an Iterable value for a Variable is assumed to be giving it a sweep value: for example, ``x = Variable("x", [1, 2, 3])`` will sweep ``x`` over three values.
+
+Solving Sweeps
+--------------
+A Model with sweeps will solve for all possible combinations: e.g., if there’s a variable ``x`` with value ``('sweep', [1, 3])`` and a variable ``y`` with value ``('sweep', [14, 17])`` then the gp will be solved four times, for :math:`(x,y)\in\left\{(1, 14),\ (1, 17),\ (3, 14),\ (3, 17)\right\}`. The returned solutions will be a one-dimensional array (or 2-D for vector variables), accessed in the usual way.
+Sweeping Vector Variables
+
+Vector variables may also be substituted for: ``y = VectorVariable(3, "y", value=('sweep' ,[[1, 2], [1, 2], [1, 2]])`` will sweep :math:`y\ \forall~y_i\in\left\{1,2\right\}`.
+
+Parallel Sweeps
+---------------
+During a normal sweep, each result is independent, so they can be run in parallel. To use this feature, run ``$ ipcluster start`` at a terminal: it will automatically start a number of iPython parallel computing engines equal to the number of cores on your machine, and when you next import gpkit you should see a note like ``Using parallel execution of sweeps on 4 clients``. If you do, then all sweeps performed with that import of gpkit will be parallelized.
+
+This parallelization sets the stage for gpkit solves to be outsourced to a server, which may be valuable for faster results; alternately, it could allow the use of gpkit without installing a solver.
+
+1D Autosweeps
+-------------
+If you're only sweeping over a single variable, autosweeping lets you specify a
+tolerance for cost error instead of a number of exact positions to solve at.
+GPkit will then search the sweep segment for a locally optimal number of sweeps
+that can guarantee a max absolute error on the log of the cost.
+
+Accessing variable and cost values from an autosweep is slightly different, as
+can be seen in this example:
+
+.. literalinclude:: examples/autosweep.py
+
+If you need access to the raw solutions arrays, the minimum tree spanning
+any given point can be gotten with ``min_bst = bst.min_bst(val)``, the extents of that tree with ``bst.bounds`` and solutions of that tree with ``bst.sols``. More information is in ``help(bst)``.
+
+
+Tight ConstraintSets
+====================
+
+Tight ConstraintSets will warn if any inequalities they contain are not
```

Comment:
"(that is, if the right side does not equal the left side)" (currently, the "that is" says the opposite from the text before it)